### PR TITLE
Solution

### PR DESCRIPTION
**Pull request by ik-web**
[DEMO LINK](https://ik-web.github.io/react_clock/)

### CODE REVIEW

File: src/components/Clock/Clock.tsx [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+};
+
+export class Clock extends React.Component<Props, State> {
+  state = {
+    time: new Date(),
+  };
+
+  timerId = setInterval(() => {
+    this.setState({ time: new Date() });
+  }, 1000);
```
Comment (vsupruniuk) [inline src/components/Clock/Clock.tsx:19]: Час потрібно друкувати також і в консолі

**Review by vsupruniuk — CHANGES_REQUESTED** [review]

**Review by vsupruniuk — APPROVED** [review]
Nice👍

File: src/App.tsx [hunk 2]
```diff
@@ -1,35 +1,65 @@
...
+  };
 
-  // This code starts a timer
-  const timerId = window.setInterval(() => {
-    // ...
-  }, 1000);
+  setRandomName = () => {
+    if (!this.state.hasClock) {
+      return;
+    }
```
Comment (illiarerun) [inline src/App.tsx:32]: you don't even use this handler
```suggestion
```

File: src/components/Clock/Clock.tsx [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+  state = {
+    time: new Date(),
+  };
+
+  timerId = setInterval(() => {
+    this.setState({ time: new Date() });
+
+    // eslint-disable-next-line no-console
+    console.log(`Time is: ${this.state.time.toLocaleTimeString()}`);
+  }, 1000);
```
Comment (illiarerun) [inline src/components/Clock/Clock.tsx:22]: you need to do it in `componentDidMount`

**Review by illiarerun — APPROVED** [review]
Approve, but pay attention to my comments:

File: src/App.tsx (hunk 2 above)
Comment (ik-web) [reply src/App.tsx]: Fixed

**Pull request closed without merging**